### PR TITLE
Fix interpose

### PR DESCRIPTION
**Pull request by Bolodya1997**
# Issue
`interpose.NewServer()` works incorrect for refresh, close scenarios. Existing test doesn't cover it.
# Solution
Remove bool switches, check _if interpose_ case with `connectionInfo.clientConnID`. Add valid test.

### CODE REVIEW

**Review by haiodo — APPROVED** [review]
Looks good.

File: pkg/networkservice/common/interpose/server_test.go [hunk 1]
```diff
@@ -14,131 +14,109 @@
...
-	})
-	require.Nil(t, err)
-	require.NotNil(t, reg)
+	// 3. Close
 
-	interposeRegName := reg.Name
-	interposeNSE := endpoint.NewServer(ctx, interposeRegName,
-		authorize.NewServer(),
-		TokenGenerator)
+	conn = conn.Clone()
```
Comment (denis-tingajkin) [inline pkg/networkservice/common/interpose/server_test.go:102]: Can we add a chain element into the end to check that Close has been called on the last chain element(NSE)

**Review by denis-tingajkin — COMMENTED** [review]
In general looks good

File: pkg/networkservice/common/interpose/server_test.go (hunk 1 above)
Comment (Bolodya1997) [reply pkg/networkservice/common/interpose/server_test.go:113]: Added `touchServer` to verify that every `Request`, `Close` is invoked on the last chain element.

**Review by denis-tingajkin — APPROVED** [review]

**Pull request merged**